### PR TITLE
Add `CUSTOMER_INTERCHANGE_DTYPES`

### PR DESCRIPTION
**Pull request by benglewis**

### CODE REVIEW

File: hirundo/dataset_optimization.py [hunk 1]
```diff
@@ -42,6 +45,26 @@ class RunStatus(Enum):
...
+    "image_path": str,
+    "label_path": str,
+    "segments_mask_path": str,
+    "segment_id": np.int32,
+    "label": str,
+    "bbox_id": str,
+    "x1": np.int32,
+    "y1": np.int32,
+    "x2": np.int32,
+    "y2": np.int32,
```
Comment (mishana) [inline hirundo/dataset_optimization.py:58]:
```suggestion
    "xmin": np.float32,
    "ymin": np.float32,
    "xmax": np.float32,
    "ymax": np.float32,
```
@shmuelyo Ishurcha

File: hirundo/_http.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import hirundo.logger
+
+logger = hirundo.logger.get_logger(__name__)
+
+
+def raise_for_status_with_reason(response: Response):
+    try:
+        response.reason = response.json().get("reason", None)
+    except Exception as e:
+        logger.debug("Failed to parse response as JSON: %s", e)
```
Comment (mishana) [inline hirundo/_http.py:12]: This won't be the output to the user, right? It's just a debug log

**Review by mishana — APPROVED** [review]
Just a few corrections. Elegant solution though :)

File: hirundo/_http.py (hunk 2 above)
Comment (benglewis) [reply hirundo/_http.py:12]: If the user has the logger set to debug, it will show to the user. The default is `INFO`

**Pull request merged**